code/backend/batch/utilities/search/azure_search_handler.py

```diff
@@ -63,6 +64,16 @@ def delete_files(self, files):
 
         return ", ".join(files_to_delete)
 
+    def delete_from_index(self, blob_url):
```

Comment:
Instead of repeating this pattern of:
1. Searching for documents
2. Transforming them
3. Deleting them
in both this and the integrated vectorisation search handler.

Could you create a `search_by_blob_url` function in each, and then call it from the `search_handler_base`? That way the pattern is reused and only the specific search part is in each handler.

WDYT?

Reply:
Updated